mava/components/jax/training/trainer.py

```diff
@@ -15,57 +15,89 @@
 
 """Trainer components for system builders."""
 
+import abc
 from dataclasses import dataclass
-from typing import Callable, Dict, List, Optional, Union
+from typing import Any, Callable, Dict, List, Optional, Union
 
 from mava.components.jax import Component
 from mava.core_jax import SystemBuilder, SystemTrainer
 from mava.utils import enums
 from mava.utils.sort_utils import sort_str_num
 
 
+class BaseTrainerInit(Component):
+    @abc.abstractmethod
+    def __init__(
+        self,
+        config: Any,
+    ):
+        """Initialise system init components.
+
+        Args:
+            config : a dataclass specifying the component parameters.
+        """
+        self.config = config
+
+    @abc.abstractmethod
+    def on_building_init_end(self, builder: SystemBuilder) -> None:
+        """Summary."""
+        pass
+
+    @abc.abstractmethod
+    def on_training_utility_fns(self, trainer: SystemTrainer) -> None:
+        """Summary."""
+        pass
+
+    @staticmethod
+    def name() -> str:
+        """Component name."""
+
+        return "trainer_init"
+
+
 @dataclass
-class TrainerInitConfig:
+class BaseTrainerInitConfig:
     trainer_networks: Union[
         Dict[str, List], enums.Trainer
     ] = enums.Trainer.single_trainer
 
 
-class TrainerInit(Component):
-    def __init__(self, config: TrainerInitConfig = TrainerInitConfig()):
-        """_summary_
+@dataclass
+class SingleTrainerInitConfig(BaseTrainerInitConfig):
+    pass
+
+
+class SingleTrainerInit(BaseTrainerInit):
+    def __init__(self, config: SingleTrainerInitConfig = SingleTrainerInitConfig()):
+        """Initialises a single trainer.
+
+        Here a single trainer is used to train all networks.
```

Comment:
small thing but I think you can remove "here a" from the comment (elsewhere too)

Reply:
Sounds good. Will do.